tests/build/_internal/config/test_config.py

```diff
@@ -24,6 +25,53 @@ def test_raises_when_trying_to_add_duplicate_config_setting(self):
         with pytest.raises(ValueError, match="lib.param already defined"):
             conf.update(prepare({"config": {"param": {"value": 0}}}, source_name="lib"))
 
+    def test_memories_includes_mbed_rom_ram_start(self):
+        input_dict = {
+            "mbed_ram_size": "0x80000",
+            "mbed_ram_start": "0x24000000",
+            "mbed_rom_size": "0x200000",
+            "mbed_rom_start": "0x08000000",
+        }
+
+        conf = Config(prepare(input_dict, source_name="lib"))
+
+        assert Memory("RAM", "lib", "0x24000000", "0x80000") in conf["memories"]
+        assert Memory("ROM", "lib", "0x8000000", "0x200000") in conf["memories"]
+
+    def test_logs_overwrite_mbed_ram_repeated(self, caplog):
+        caplog.set_level(logging.DEBUG)
+        input_dict = {"mbed_ram_size": "0x80000", "mbed_ram_start": "0x24000000"}
+
+        conf = Config(prepare(input_dict, source_name="lib1"))
+        conf.update(prepare(input_dict, source_name="lib2"))
+
+        assert "values from `lib2` will be ignored" in caplog.text
+
+    def test_memory_attr_converted_as_hex(self):
+        input_dict = {"mbed_ram_size": "1024", "mbed_ram_start": "0x24000000"}
+
+        conf = Config(prepare(input_dict, source_name="lib"))
+
+        assert Memory("RAM", "lib", "0x24000000", "0x400") in conf["memories"]
+
+    def test_raises_memory_attr_not_integer(self):
```

Comment:
The tests for `prepare` should probably be in `test_source.py`. This test suite is supposed to check the `Config` object behaves correctly. We only used `prepare` to format the input data for `Config` here previously,  now we're also testing `prepare` directly.